src/pudl/transform/eia861.py

```diff
@@ -1288,6 +1252,148 @@ def distributed_generation(tfr_dfs):
+    ]
+
+    tech_cols = [
+        'all_storage_capacity_mw',
+        'combustion_turbine_capacity_mw',
+        'combustion_turbine_capacity_pct',
+        'estimated_or_actual_tech_data',
+        'hydro_capacity_mw',
+        'hydro_capacity_pct',
+        'internal_combustion_capacity_mw',
+        'internal_combustion_capacity_pct',
+        'other_capacity_mw',
+        'other_capacity_pct',
+        'pv_capacity_mw',
+        'steam_capacity_mw',
+        'steam_capacity_pct',
+        'total_capacity_mw',
+        'wind_capacity_mw',
+        'wind_capacity_pct',
+    ]
+
+    fuel_cols = [
+        'oil_fuel_pct',
+        'estimated_or_actual_fuel_data',
+        'gas_fuel_pct',
+        'other_fuel_pct',
+        'renewable_fuel_pct',
+        'water_fuel_pct',
+        'wind_fuel_pct',
+        'wood_fuel_pct',
+    ]
+
+    # Pre-tidy transform: set estimated or actual A/E values to 'Acutal'/'Estimated'
+    raw_dg = (
+        tfr_dfs['distributed_generation_eia861'].copy()
+        .assign(
+            estimated_or_actual_capacity_data=lambda x: (
+                x.estimated_or_actual_capacity_data.map(pc.ESTIMATED_OR_ACTUAL)),
+            estimated_or_actual_fuel_data=lambda x: (
+                x.estimated_or_actual_fuel_data.map(pc.ESTIMATED_OR_ACTUAL)),
+            estimated_or_actual_tech_data=lambda x: (
+                x.estimated_or_actual_tech_data.map(pc.ESTIMATED_OR_ACTUAL))
+        )
+    )
+
+    # Split into three tables: Capacity/tech-related, fuel-related, and misc.
+    raw_dg_tech = raw_dg[idx_cols + tech_cols].copy()
+    raw_dg_fuel = raw_dg[idx_cols + fuel_cols].copy()
+    raw_dg_misc = raw_dg[idx_cols + misc_cols].copy()
+
+    ###########################################################################
+    # Transform Values:
+    # * Turn pct values into mw values
+    # * Remove old pct cols and totals cols
+    ###########################################################################
+
+    # Separate datasets into years with only pct values (pre-2010) and years with only mw values (post-2010)
+    df_pre_2010_tech = raw_dg_tech[raw_dg_tech['report_date'] < '2010-01-01']
+    df_post_2010_tech = raw_dg_tech[raw_dg_tech['report_date'] >= '2010-01-01']
+    df_pre_2010_misc = raw_dg_misc[raw_dg_misc['report_date'] < '2010-01-01']
+    df_post_2010_misc = raw_dg_misc[raw_dg_misc['report_date'] >= '2010-01-01']
+
+    logger.info(
+        'Converting pct values into mw values for distributed generation misc table')
+    transformed_dg_misc = (
+        df_pre_2010_misc.assign(
+            distributed_generation_owned_capacity_mw=lambda x: _pct_to_mw(
+                x, 'distributed_generation_owned_capacity_pct'),
+            backup_capacity_mw=lambda x: _pct_to_mw(x, 'backup_capacity_pct'),
+        ).append(df_post_2010_misc)
+        .drop(['distributed_generation_owned_capacity_pct',
+               'backup_capacity_pct',
+               'total_capacity_mw'], axis=1)
```

Comment:
why are these columns being dropped here?

Reply:
When a table goes through the `_pct_to_mw()` function to calculate the **mw** component breakdown from the **pct** value. it does not delete the old, now irrelevant `_pct` columns. This is why these columns are deleted here, after running `_pct_to_mw()` instead. The `total_capacity_mw` col is also dropped here because, as a total column, it is not included in the final output. This is the `_misc` table and only handles the `distributed_generation_owned_capacity` and `backup_capacity_pct` columns because they are not considered "components" in the same way that `hydro_capacity_pct` etc. are (this was the issue with the totals not equaling the sum). Below, there are a few lines of code that do the same thing for the `_tech` table, housing these other "true" component columns and deleting their `_pct` column counterparts after running `pct_to_mw()`